docs/security/cors.rst

```diff
@@ -1,8 +1,311 @@
+	:emphasize-lines: 3
+
+The second approach is to define one or more named CORS policies, and then select the policy by name at run time.
+
+.. literalinclude:: cors/sample/src/CorsExample2/Startup.cs
+	:language: csharp
+	:lines: 9-18,20-23,27
+	:dedent: 8
+
+This example adds a CORS policy named "AllowSpecificOrigin". To select the policy, pass the name to UseCors.  
+
+.. _cors-policy-options:
+
+CORS policy options
+-------------------
+		
+This section describes the various options that you can set in a CORS policy.	
+
+    - `Set the allowed origins`_
+    - `Set the allowed HTTP methods`_
+    - `Set the allowed request headers`_
+    - `Set the exposed response headers`_
+    - `Credentials in cross-origin requests`_	
+    - `Set the preflight expiration time`_
+
+For some options it may be helpful to read `How CORS works`_ first.
+    
+Set the allowed origins
+^^^^^^^^^^^^^^^^^^^^^^^
+
+To allow one or more specific origins:
+
+.. literalinclude:: cors/sample/src/CorsExample4/Startup.cs
+	:language: csharp
+	:start-after: BEGIN01
+	:end-before: END01
+	:dedent: 16
+	
+To allow all origins:
+
+.. literalinclude:: cors/sample/src/CorsExample4/Startup.cs
+	:language: csharp
+	:start-after: BEGIN02
+	:end-before: END02
+	:dedent: 16
+
+Consider carefully before allowing requests from any origin. It means that literally any website can make AJAX calls to your app.
+
+Set the allowed HTTP methods
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To specify which HTTP methods are allowed to access the resource. 
+
+.. literalinclude:: cors/sample/src/CorsExample4/Startup.cs
+	:language: csharp
+	:start-after: BEGIN03
+	:end-before: END03
+	:dedent: 16
+					
+To allow all HTTP methods:
+					
+.. literalinclude:: cors/sample/src/CorsExample4/Startup.cs
+	:language: csharp
+	:start-after: BEGIN04
+	:end-before: END04
+	:dedent: 16
+
+This affects pre-flight requests and Access-Control-Allow-Methods header. 
+
+Set the allowed request headers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A CORS preflight request might include an Access-Control-Request-Headers header, listing the HTTP headers set by the application (the so-called "author request headers"). To whitelist specific headers:
+
+.. literalinclude:: cors/sample/src/CorsExample4/Startup.cs
+	:language: csharp
+	:start-after: BEGIN05
+	:end-before: END05
+	:dedent: 16
+
+To allow all author request headers:
+
+.. literalinclude:: cors/sample/src/CorsExample4/Startup.cs
+	:language: csharp
+	:start-after: BEGIN06
+	:end-before: END06
+	:dedent: 16
+
+Browsers are not entirely consistent in how they set Access-Control-Request-Headers. If you set headers to anything other than "*", you should include at least "accept", "content-type", and "origin", plus any custom headers that you want to support. 
+
+Set the exposed response headers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, the browser does not expose all of the response headers to the application. The response headers that are available by default are:
```

Comment:
a link to the RFC might be helpful http://www.w3.org/TR/cors/#simple-response-header